src/Simulation/Core/Attributes.cs

```diff
@@ -6,6 +6,23 @@
 
 namespace Microsoft.Quantum.Simulation.Core
 {
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+    public class SpecializationRangeAttribute : Attribute
+    { 
+        public string SourceFile { get; }
+        public string SpecializationKind { get; }
```

Comment:
Sorry for not pointing out this in earlier discussion, but there is `enum`
https://github.com/microsoft/qsharp-runtime/blob/ebb4b2a9de0110ab45a6e38b38cfbd664759483f/src/Simulation/Core/OperationFunctor.cs#L9 
that might be useful here.

Reply:
changed.